src/components/MazeMap/MazeMap.tsx

```diff
@@ -131,17 +133,74 @@ const MazeMap = (props: MazeMapProps) => {
   const prepareMap = () => {
     if (window.Mazemap) {
       const map = new window.Mazemap.Map(mapOptions);
-      map.on('load', () => {
+      map.on("load", () => {
         if (props.controls) {
           map.addControl(new window.Mazemap.mapboxgl.NavigationControl());
         }
+        if (props.hideIcons && (!props.hideAll)) {
+          let style = map.getStyle();
+
+          // layer 143 room icons
+          let layer143 = style.layers[143];
+          // layer144 building icons
+          let layer144 = style.layers[144];
+
+          layer143.layout = {
```

Comment:
could you pls add some comments that explains what's going on here and how it hides the icons?